src/frequenz/sdk/data_ingestion/resampling/timeseries_resampler.py

```diff
@@ -0,0 +1,118 @@
+"""
+TimeSeriesResampler class that delegates resampling to individual resmaplers.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+import logging
+from datetime import datetime
+from typing import Callable, Dict, Optional, Sequence
+
+import pytz
+
+from frequenz.sdk.core.timeseries import Sample, TimeSeriesId
+from frequenz.sdk.data_ingestion.resampling.individual_timeseries_resampler import (
+    IndividualTimeSeriesResampler,
+)
+
+logger = logging.Logger(__name__)
+
+ResamplingFunction = Callable[[Sequence[Sample], float], float]
+
+DEFAULT_BUFFER_SIZE = 100
+
+
+class TimeSeriesResampler:
+    """TimeSeriesResampler class that delegates resampling to individual resmaplers."""
+
+    def __init__(
+        self,
+        period_s: float,
+        max_data_age_in_periods: float,
+        resampling_function: ResamplingFunction,
+    ) -> None:
+        """Initialize the TimeSeriesResampler.
+
+        Args:
+            period_s: resampling frequency, in seconds
+            max_data_age_in_periods: max age that samples shouldn't exceed in order
+                to be used in the resampling function
+            resampling_function: function to be applied to a sequence of samples within
+                a resampling period to produce a single output sample
+        """
+        self._period_s = period_s
+        self._max_data_age_in_periods: float = max_data_age_in_periods
+        self._resampling_function: ResamplingFunction = resampling_function
+        self._resamplers: Dict[TimeSeriesId, IndividualTimeSeriesResampler] = {}
+
+    def add_time_series(self, time_series_id: TimeSeriesId) -> None:
+        """Create a new resampler for a specific time series.
+
+        Args:
+            time_series_id: time series id
+
+        Raises:
+            ValueError: if resampler for the provided timer_series_id already exists
+        """
+        resampler = self._resamplers.get(time_series_id)
+        if resampler is None:
+            raise ValueError(
+                f"Resampler for time series {time_series_id} already exists!"
+            )
+
+        self._resamplers[time_series_id] = IndividualTimeSeriesResampler(
+            buffer_size=DEFAULT_BUFFER_SIZE,
```

Comment:
I will make it configurable via a constructor param.

Reply:
Yeah, the idea of having `max_data_age_in_periods` was to calculate the buffer size from that and the `period_s`, because the idea is that it will be adjusted automatically in the future, based on the input sample rate.

Reply:
> I will make it configurable via a constructor param.

No, please infer it from the `max_data_age_in_periods` and `period_s`. This will need to be adjusted dynamically.

Reply:
As I said before, I half take back that comment and would be fine to pass a `buffer_len` directly, or even hard-code it for now in the Resampler for this PR.

Reply:
As discussed, the buffer size is not constraint, but when resampling happens, the outdated samples will be removed from the buffer. (samples older than `now() - timedelta(seconds=max_data_age_in_periods * resampling_period_s)`)